[0.71] Add page on React Native Gradle Plugin for 71 (#3476)
facebook/react-native-website: docs/react-native-gradle-plugin.md

### `bundleConfig`

This is the path to a configuration file that will be passed to `bundle --config <file>` if provided. Default is empty (no config file will be probided). More information on bundling config files can be found [on the CLI documentation](https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle). Can be customized as follow:

```groovy
bundleConfig = file(../rn-cli.config.js)
```

### `bundleAssetName`

This is the name of the bundle file that should be generated. Default is `index.android.bundle`. Can be customized as follow:

```groovy
bundleAssetName = "MyApplication.android.bundle"
```

### `entryFile`

The entry file used for bundle generation. The default is to search for `index.android.js` or `index.js`. Can be customized as follow:

```groovy
entryFile = file("../js/MyApplication.android.js")
```

### `extraPackagerArgs`

A list of extra flags that will be passed to the `bundle` command. The list of available flags is in [the CLI documentation](https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle). Default is empty. Can be customized as follows:

```groovy
extraPackagerArgs = []
```

### `hermesCommand`

> The path to the `hermesc` command (the Hermes Compiler). React Native comes with a version of the Hermes compiler bundled it with, so you generally won't be needing to customize this, by default the plugin will use the correct compiler for your system.

Reviewer: ```suggestion
The path to the `hermesc` command (the Hermes Compiler). React Native comes with a version of the Hermes compiler bundled with it, so you generally won't be needing to customize this. The plugin will use the correct compiler for your system by default.
```

If the plugin uses the right compiler, why do we have this key? Are there setups that requires its customization?

```

## Using Flavors & Build Variants

When building Android apps, you might want to use [custom flavors](https://developer.android.com/studio/build/build-variants#product-flavors) to have different versions of your app starting from the same project.

Please refer to the [official Android guide](https://developer.android.com/studio/build/build-variants) to configure custom build types (like `staging`) or custom flavors (like `full`, `lite`, etc.).
By default new apps are create with two build types (`debug` and `release`) and no custom flavors.

The combination of all the build types and all the flavors generates a set of **build variants**. For instance for `debug`/`staging`/`release` build types and `full`/`lite` you will have 6 build variants: `fullDebug`, `fullStaging`, `fullRelease` and so on.

If you're using custom variants beyond `debug` and `release`, you need to instruct the React Native Gradle Plugin specifying which of your variants are **debuggable** using the [`debuggableVariants`](#debuggablevariants) configuration as follows:

```diff
apply plugin: "com.facebook.react"

react {
+ debuggableVariants = ["fullStaging", "fullDebug"]
}
```

> This is necessary as for `debuggableVariants`, the plugin will skip the JS bundling, so you'll need Metro to run them. So if you list `fullStaging` in the `debuggableVariants`, you won't be able to publish it to a store as it will be missing the bundle.

Reviewer: ```suggestion
This is necessary because the plugin will skip the JS bundling for all the `debuggableVariants`: you'll need Metro to run them. For example, if you list `fullStaging` in the `debuggableVariants`, you won't be able to publish it to a store as it will be missing the bundle.
```

> ## What is the plugin doing under the hood?

Reviewer: Does it make sense to move this section above. It feels like a `why` that we want to tell to the users **before** explaining all the possible configurations.

What do you think?

Author: I was actually thinking to move this to a separate page inside the Architecture -> Build Logic section where we have also the Bundle Hermes docs as this is not necessarily a guide.

Author: I'll keep it here for now, but I'll move it to architecture as I plan to add a couple of sections in the future 👍

> The React Native Gradle Plugin is responsible of configuring your Application build so that you can ship React Native applications to production.

Reviewer: ```suggestion
The React Native Gradle Plugin is responsible for configuring your Application build to ship React Native applications to production.
```
